src/inspector_js_api.cc

```diff
@@ -32,6 +32,171 @@ std::unique_ptr<StringBuffer> ToProtocolString(Isolate* isolate,
   return StringBuffer::create(StringView(*buffer, buffer.length()));
 }
 
+class JSDispatcher;
+
+class JSDispatcherInterface : private AsyncWrap {
```

Comment:
Instead of having three top-level classes, can you reorganize it so that it's clearer to the reader that these three concepts in fact form one cohesive module? Maybe something like

```cpp
class JSDispatcher : public MessageDispatcher {
 public:
  class Factory : public MessageDispatcherFactory {};
 private:
  class Interface : private AsyncWrap {};
  JSDispatcher(std::unique_ptr<DispatcherSession>&& session,
               Interface* interface) {}
};
```

and then later

```cpp
new JSDispatcher::Factory()
```

In fact if you do so you might be able to get rid of a few accessors too.